### DISCUSSION TITLE
Secure website locally without appending TLD

### DISCUSSION THREAD

**Discussion by bzrncev** [beyondcode/herd-community#479] [Ideas]
I have a local site that needs to run as `local.example.com`, which I've added to the `hosts` file. Securing this site with the default configurations of Valet or Herd is not possible since they append the `.test` TLD to the URL.

However, by disabling the following if check in Valet's `Site.php` file, and returning the $domain, it's possible to secure the site without the `.test` TLD:

```php
// Don't add .TLD if user already passed the string in with the TLD on the end
if ($domain && str_contains($domain, '.'.$this->config->read()['tld'])) {
    return $domain;
}
```

I'm wondering if it's possible to have a similar option in Herd, or the code is bundled in a way that makes such modifications difficult. I understand this is an edge case feature that only a small number of users would need, but it would be great if you could consider adding an option to disable appending .test TLD for specific domains in a future Herd release if someone doesn't need it.

**Comment by khawarmd** [comment]
I also follow a similar routine; I initiate my local websites with a "local." prefix. This strategy is beneficial when accessing the live website, as browser recommendations might sometimes direct you to the local site, even though you intended to visit the live version.
